paimon-flink/paimon-flink1-common/src/main/java/org/apache/flink/table/data/RowData.java

```diff
@@ -0,0 +1,314 @@
+ * </pre>
+ *
+ * <p>Nullability is always handled by the container data structure.
+ */
+@PublicEvolving
+public interface RowData {
+
+    /**
+     * Returns the number of fields in this row.
+     *
+     * <p>The number does not include {@link RowKind}. It is kept separately.
+     */
+    int getArity();
+
+    /**
+     * Returns the kind of change that this row describes in a changelog.
+     *
+     * @see RowKind
+     */
+    RowKind getRowKind();
+
+    /**
+     * Sets the kind of change that this row describes in a changelog.
+     *
+     * @see RowKind
+     */
+    void setRowKind(RowKind kind);
+
+    // ------------------------------------------------------------------------------------------
+    // Read-only accessor methods
+    // ------------------------------------------------------------------------------------------
+
+    /** Returns true if the field is null at the given position. */
+    boolean isNullAt(int pos);
+
+    /** Returns the boolean value at the given position. */
+    boolean getBoolean(int pos);
+
+    /** Returns the byte value at the given position. */
+    byte getByte(int pos);
+
+    /** Returns the short value at the given position. */
+    short getShort(int pos);
+
+    /** Returns the integer value at the given position. */
+    int getInt(int pos);
+
+    /** Returns the long value at the given position. */
+    long getLong(int pos);
+
+    /** Returns the float value at the given position. */
+    float getFloat(int pos);
+
+    /** Returns the double value at the given position. */
+    double getDouble(int pos);
+
+    /** Returns the string value at the given position. */
+    StringData getString(int pos);
+
+    /**
+     * Returns the decimal value at the given position.
+     *
+     * <p>The precision and scale are required to determine whether the decimal value was stored in
+     * a compact representation (see {@link DecimalData}).
+     */
+    DecimalData getDecimal(int pos, int precision, int scale);
+
+    /**
+     * Returns the timestamp value at the given position.
+     *
+     * <p>The precision is required to determine whether the timestamp value was stored in a compact
+     * representation (see {@link TimestampData}).
+     */
+    TimestampData getTimestamp(int pos, int precision);
+
+    /** Returns the raw value at the given position. */
+    <T> RawValueData<T> getRawValue(int pos);
+
+    /** Returns the binary value at the given position. */
+    byte[] getBinary(int pos);
+
+    /** Returns the array value at the given position. */
+    ArrayData getArray(int pos);
+
+    /** Returns the map value at the given position. */
+    MapData getMap(int pos);
+
+    /**
+     * Returns the row value at the given position.
+     *
+     * <p>The number of fields is required to correctly extract the row.
+     */
+    RowData getRow(int pos, int numFields);
+
+    /** Returns the variant value at the given position. */
+    Variant getVariant(int pos);
```

Comment:
Do we need to add a RowData interface with getVariant method to paimon-flink-2.0 as well?

Reply:
I think it should, but it feels off that this is not caught by the CI. I will have a look at the CI.

Reply:
It is added. And the e2e test for flink 2.0 is added to cover the situation.